Emby.Server.Implementations/ScheduledTasks/TaskManager.cs

```diff
@@ -25,20 +28,23 @@ public class TaskManager : ITaskManager
 
         private readonly IApplicationPaths _applicationPaths;
         private readonly ILogger<TaskManager> _logger;
+        private readonly IServiceProvider _serviceProvider;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="TaskManager" /> class.
         /// </summary>
         /// <param name="applicationPaths">The application paths.</param>
         /// <param name="logger">The logger.</param>
+        /// <param name="serviceProvider">The service provider.</param>
         public TaskManager(
             IApplicationPaths applicationPaths,
-            ILogger<TaskManager> logger)
+            ILogger<TaskManager> logger,
+            IServiceProvider serviceProvider)
```

Comment:
Do we really need to pull in the full service provider just to later get the lilbrary manager?

Reply:
ILibraryManager injects ITaskManager creating a ciruclar dependecy otherwise. Without factories i dont know how to get around that.

Reply:
Correct, but that would as i said just create a factory and move the usage of the serviceprovider somewhere else ^^

Reply:
Yes, but it makes the dependencies of this class more clear and is also makes a cleaner implementation if you need to mock the dependency for testing